### PR TITLE
allow extra volumes to be mounted

### PR DESCRIPTION
**Pull request by mikeraimondi**
Signed-off-by: Michael Raimondi <mike@mikeraimondi.com>

#### What this PR does / why we need it:
Allows extra volumes to be mounted into Vault. Very handy for plugins.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by jbialy** [comment]
I would think that `vault.customSecrets` should really just be an extra mount defined in `vault.extraVolumeMounts` 😄

**Comment by mikeraimondi** [comment]
@jbialy are you suggesting that `customSecrets` be removed? I can see how that would simplify the chart slightly, at the cost of slightly more verbose configuration.

**Comment by jbialy** [comment]
> @jbialy are you suggesting that `customSecrets` be removed? I can see how that would simplify the chart slightly, at the cost of slightly more verbose configuration.

I was thinking that `customSecrets` seems to be very specific to just vault's TLS keys, where it's nothing more than another volume mount? 

Maybe all that's needed is a comment that outlines how to include the TLS keys as part of `extraVolumeMounts`?

**Comment by mikeraimondi** [comment]
@jbialy good idea! I made the change.

**Comment by jbialy** [comment]
> @jbialy good idea! I made the change.

Looking at it now, I can see some advantage to having `vault.customSecrets` as it does take care of creating both the volume and mount point for you (pretty much was your said about this being more verbose). Never-mind the fact that the underlying change will likely brake existing deployments 🤔. Let's see what the approvers say. Apologies if I confused things for no reason!

**Comment by lachie83** [comment]
cc @jpds

File: incubator/vault/values.yaml
Comment (lachie83) [inline incubator/vault/values.yaml:136]: This looks like a breaking change as it's removing existing fields. Is there a way we could keep the existing field otherwise I'm not sure this is a patch level change and it might constitute a minor version bump.

Comment (mikeraimondi) [reply incubator/vault/values.yaml:136]: A [previous iteration of this PR](https://github.com/helm/charts/pull/9250/commits/f4a66018f4ef354d81d5201ae3cac36f40bf1146) added extraVolumeMounts without removing customSecrets. I can revert to that commit if you think it would stand a better chance of getting merged.

Comment (lachie83) [reply incubator/vault/values.yaml:136]: I think this has implications for rolling back to a previous chart version. I would like to see what the chart maintainer would like to do here. cc @jpds

Comment (lachie83) [reply incubator/vault/values.yaml:136]: It might be okay to bump the minor version in this case

Comment (mikeraimondi) [reply incubator/vault/values.yaml:136]: @lachie83 I bumped the minor version. Thanks for the feedback!

**Comment by lachie83** [comment]
paging @jpds as for approval

**Comment by sdickhoven** [comment]
is there a way to get this pr merged? we're using vault with google 2fa and need the custom volumes to configure the google auth plugin for vault.

**Comment by mikeraimondi** [comment]
@lachie83 @jpds latest merge conflicts are resolved! Any hope of getting this merged in?

**Comment by mikeraimondi** [comment]
@jpds @mattfarina this PR should be up-to-date and ready for review. Thanks!

**Comment by mattfarina** [comment]
There is a vault chart in the Helm hub that continues to use custom secrets at https://hub.helm.sh/charts/banzaicloud-stable/vault. It appears this chart has its lineage from the chart in helm/charts. An option still exists for people who want to carry that on.

@mikeraimondi For those using the `customSecrets` could you write a short update note for the readme.

If @jpds doesn't approve/reject then I plan on approving when the the readme is updated.

**Comment by mikeraimondi** [comment]
@mattfarina I updated the readme. Let me know if the changes look good.

**Review by mattfarina — APPROVED** [review]

**Pull request merged**
